pktoken/osm.go

```diff
@@ -0,0 +1,88 @@
+package pktoken
+
+import (
+	"crypto"
+	"fmt"
+
+	"github.com/lestrrat-go/jwx/v2/jws"
+)
+
+func (p *PKToken) NewSignedMessage(content []byte, signer crypto.Signer) ([]byte, error) {
+	alg, _, _, err := p.GetCicValues()
+	if err != nil {
+		return nil, err
+	}
+
+	pktHash, err := p.Hash()
+	if err != nil {
+		return nil, err
+	}
+
+	// Create our headers as defined by section 3.5 of the OpenPubkey paper
+	protected := jws.NewHeaders()
+	if err := protected.Set("alg", alg); err != nil {
+		return nil, err
+	}
+	if err := protected.Set("kid", pktHash); err != nil {
+		return nil, err
+	}
+	if err := protected.Set("typ", "osm"); err != nil {
+		return nil, err
+	}
+
+	return jws.Sign(
+		content,
+		jws.WithKey(
+			alg,
+			signer,
+			jws.WithProtectedHeaders(protected),
+		),
+	)
+}
+
+func (p *PKToken) VerifySignedMessage(osm []byte) ([]byte, error) {
+	alg, _, upk, err := p.GetCicValues()
+	if err != nil {
+		return nil, err
+	}
+
+	message, err := jws.Parse(osm)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check that our OSM headers are correct
+	if len(message.Signatures()) != 1 {
+		return nil, fmt.Errorf("expected only one signature on jwt, received %d", len(message.Signatures()))
+	}
+	protected := message.Signatures()[0].ProtectedHeaders()
+
+	// Verify typ header matches expected "osm" value
+	if typ, ok := protected.Get("typ"); !ok {
+		return nil, fmt.Errorf("missing required header `typ`")
+	} else if typ != "osm" {
+		return nil, fmt.Errorf(`incorrect "typ" header, expected "osm" but recieved %s`, typ)
+	}
```

Comment:
sounds perfectly good to me. I can get self conscious about how long my code is but I don't have a preference otherwise.

I also did it for the following check statements ed9493d